src/agent/onefuzz-supervisor/src/commands.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::auth::Secret;
+use anyhow::Result;
+use std::process::Stdio;
+use tokio::{fs, io::AsyncWriteExt, process::Command};
+
+#[cfg(target_os = "windows")]
+use std::{env, path::PathBuf};
+
+#[cfg(target_os = "linux")]
+use users::{get_current_uid, get_user_by_name, get_user_by_uid, os::unix::UserExt};
+
+#[derive(Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct SshKeyInfo {
+    pub key: Secret<String>,
+    pub user: Option<String>,
+    set_permissions: bool,
+}
+
+#[cfg(target_os = "windows")]
+pub async fn add_ssh_key(key_info: SshKeyInfo) -> Result<()> {
+    if key_info.user.is_some() {
+        bail!("specifying a user is not supported on windows at this time");
+    }
+
+    let mut ssh_path =
+        PathBuf::from(env::var("ProgramData").unwrap_or_else(|_| "c:\\programdata".to_string()));
+    ssh_path.push("ssh");
+
+    let mut dsa_path = ssh_path.clone();
+    let mut key_path = ssh_path;
+
+    dsa_path.push("ssh_host_dsa_key");
+    key_path.push("administrators_authorized_keys");
+
+    {
+        let mut file = fs::OpenOptions::new().append(true).open(&key_path).await?;
+        file.write_all(key_info.key.expose_ref().as_bytes()).await?;
+    }
+
+    if key_info.set_permissions {
+        verbose!("removing Authenicated Users permissions to ssh key");
+        let result = Command::new("icacls.exe")
+            .arg(&key_path)
+            .arg("/remove")
+            .arg("NT AUTHORITY/Authenticated Users")
+            .stdin(Stdio::null())
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn()?
+            .wait_with_output()
+            .await?;
+        if !result.status.success() {
+            bail!(
+                "set authorized_keys ({}) permissions failed: {:?}",
+                key_path.display(),
+                result
+            );
+        }
+
+        verbose!("removing inheritance");
```

Comment:
Could you add comments describing why we need to do each of these things? Brief comments with documentation links would be great.